### ISSUE TITLE
Bot code restructuring suggestion

### ISSUE THREAD

**Issue by user4752** [SocksTheWolf/AntiScamBot#53]
After hearing you state you wished to reorganize the code, I was reminded of a feature discordjs has and read up on the docs for discordpy and was happy to find out it does also implement a similar feature.

Discordpy [extensions](https://discordpy.readthedocs.io/en/stable/ext/commands/extensions.html#primer) allow separating out sections of code into sub-modules as you had indicated you thought about doing, and allow for hot reloading and loading of new code without needing to shut the bot down in order to do so (assuming the main bot process has been coded in a sufficient manner to allow this)

Discordpy [cogs](https://discordpy.readthedocs.io/en/stable/ext/commands/cogs.html#ext-commands-cogs) allow grouping of functions into a shared class space inside of modules which provide skeleton for a framework to run commands and bind listeners to event emitted from the discord api. 

Cogs can be the basis for organizing of slash commands into groups.
Cogs would also simplify parameter passing between classes and functions as cogs classes can be referenced externally via the `get_cog()` function call.

**Comment by SocksTheWolf** [comment]
That could be kinda nice, BotMain is getting really bloated with the number of base class references it currently has. I would love to see it more lean if possible.

**Comment by user4752** [comment]
Once #55 has been reviewed discussion on how commands can be logically organized can occur in order to further development in this idea.

I have created a sample bot using extensions and cogs have found they're extremely easy to setup, deciding on how to group commands is the difficult part. 

Next steps on testing for me are
- hot reload of commands for when updates are pushed without restarting the bot
- passing data around the cogs without needing function parameters or global variables

**Comment by SocksTheWolf** [comment]
Yeah, I think reorganizing via cogs would be nice. Some of these classes are so big it can be hard to find things. I don't think necessarily we need to do a full on refactor though, but reorg would be good.
